cumulus/client/cirrus-executor/src/worker.rs

```diff
@@ -201,6 +207,7 @@ async fn handle_slot_notifications<PBlock, PClient, BundlerFn, SecondaryHash>(
 
 async fn handle_block_import_notifications<PBlock, PClient, ProcessorFn, SecondaryHash>(
 	primary_chain_client: &PClient,
+	best_secondary_number: NumberFor<PBlock>,
```

Comment:
Why does `best_secondary_number` have type of the primary chain block number? I do not understand this and conversion above.

Reply:
Do we have other places with similar conversions? I thought we avoided that somehow until this point.

Either way, when we're doing conversion I feel like we should go through `u32`/`BlockNumber` from primitives instead of encoding/decoding it, though both will certainly work.

Reply:
Yes, we have a few places like this, there are 5 results by searching for `Primary number and secondary number must use the same type`.

I don't have a strong opinion if you want to first convert `NumberFor<Block/PBlock>` to the `u32`/`BlockNumber` and then convert this concrete type to `NumberFor<PBlock/Block>`.

Reply:
Not sure if https://github.com/subspace/subspace/pull/627/commits/83342a944f30c663ef39c4c29c3f4b3fff2d78e7 is enough, had to change a few more to make it compile, let me know if you have a better idea.

Reply:
I would rather use `TryInto` with `.expect()`, I feel like saturated conversion is often used as a way to make compiler not complain and to potentially silence actual bugs, I'd rather prefer to have explicit statements about that. And you should be able to turn `BlockNumber` (which is `u32`) into any `NumberFor<T>` without saturated conversion or anything like that.

Reply:
Updated.